internal/npm_install/generate_build_file.js

```diff
@@ -485,13 +485,13 @@ function findPackages(p = 'node_modules') {
   packages.forEach(
       f => pkgs.push(parsePackage(f), ...findPackages(path.posix.join(f, 'node_modules'))));
 
-  addDynamicDependencies(pkgs);
-
   const scopes = listing.filter(f => f.startsWith('@'))
                      .map(f => path.posix.join(p, f))
                      .filter(f => isDirectory(f));
   scopes.forEach(f => pkgs.push(...findPackages(f)));
 
+  addDynamicDependencies(pkgs);
```

Comment:
@alexeagle Related to that: Should we consider in a follow-up to just call this _once_  when the packages are determined? It looks like it re-iterates through all packages a couple of times right now?

Reply:
Yes, agreed

@soldair is looking at this code right now, getting some better tests around it, converting to TS. So I think we should revisit performance mistakes like that after he's done

Reply:
sounds good. thanks for clarifying that!